### DISCUSSION TITLE
General Question: Compilation Time >15 Minutes?

### DISCUSSION THREAD

**Discussion by mtailordev** [zephyrproject-rtos/zephyr#44554] [Q&A]
Hey guys, I have some general questions about compilation time of the Zephyr RTOS, but I haven't worked with it myself yet and have not found anything related to this online.

**My colleague claims that every compilation he does takes at least 15 minutes to complete.
Is this really the case, and if so, isn't there a way to only re-compile changed code (Not the entire stack)?**

I don't know his exact hardware + software configuration he is using (also unsure how relevant that is if you got atleast a somewhat recent system), he has some 4th-6th gen 4-core Intel CPU, >8 Gigs of RAM, Win10 and is using the Standard Zephyr sdk toolchain.

**Comment by carlescufi** [comment]
What Host operating system is your colleague using (including whether he's using WSL or MSYS or MINGW or similar)? which application/sample? which board?

**Reply by mtailordev** [reply]
Windows 10, as stated above, I don't know any more specifics about the toolchain sadly, but it is a custom board with a NRF52832 mcu.

**Comment by stephanosio** [comment]
> My colleague claims that every compilation he does takes at least 15 minutes to complete.

If that is really the case, he probably needs a new computer.

> isn't there a way to only re-compile changed code (Not the entire stack)?

That is the default behaviour.

**Reply by stephanosio** [reply]
FWIW, disabling Microsoft Defender (or any other anti-virus software for that matter) "real-time protection" setting should significantly speed things up; though, 15 minutes even with the real-time protection on is still excessive.

**Reply by mtailordev** [reply]
I've also heard certain anti-virus software can slow down compilation time significantly. As far as I know all relevant folders and files _should_ already be excluded by him..

> That is the default behaviour.

As in default = recompiling entire stack?

**Reply by stephanosio** [reply]
recompiling only changed code
